Description: Disable "Check for Updates" action.
Bug-Ubuntu: https://bugs.launchpad.net/bugs/272212
Author: Daniel Hahler <ubuntu@thequod.de>, Felix Geyer <debfx-pkg@fobos.de>

diff -Nur virtualbox-ose-4.0.0-dfsg.orig/src/VBox/Frontends/VirtualBox/src/globals/VBoxGlobal.cpp virtualbox-ose-4.0.0-dfsg/src/VBox/Frontends/VirtualBox/src/globals/VBoxGlobal.cpp
--- virtualbox-ose-4.0.0-dfsg.orig/src/VBox/Frontends/VirtualBox/src/globals/VBoxGlobal.cpp	2010-12-17 16:14:24.000000000 +0100
+++ virtualbox-ose-4.0.0-dfsg/src/VBox/Frontends/VirtualBox/src/globals/VBoxGlobal.cpp	2011-01-01 23:55:23.240440031 +0100
@@ -4565,6 +4565,9 @@
  */
 void VBoxGlobal::showUpdateDialog (bool aForce)
 {
+    // Updates are handled through package managers.
+    return;
+
     /* Silently check in one day after current time-stamp */
     QTimer::singleShot (24 /* hours */   * 60   /* minutes */ *
                         60 /* seconds */ * 1000 /* milliseconds */,
--- VirtualBox-4.1.2_OSE/src/VBox/Frontends/VirtualBox/src/runtime/UIMachineMenuBar.cpp.orig	2011-08-15 14:28:35.000000000 +0200
+++ VirtualBox-4.1.2_OSE/src/VBox/Frontends/VirtualBox/src/runtime/UIMachineMenuBar.cpp	2011-08-16 23:27:21.954256054 +0200
@@ -275,13 +275,6 @@
 #if defined(Q_WS_MAC) && (QT_VERSION < 0x040700)
     if (m_fIsFirstTime)
 # endif
-        pMenu->addAction(gActionPool->action(UIActionIndex_Simple_Update));
-#ifndef Q_WS_MAC
-    pMenu->addSeparator();
-#endif /* !Q_WS_MAC */
-#if defined(Q_WS_MAC) && (QT_VERSION < 0x040700)
-    if (m_fIsFirstTime)
-# endif
         pMenu->addAction(gActionPool->action(UIActionIndex_Simple_About));
 
 
diff -Nur virtualbox-ose-4.0.0-dfsg.orig/src/VBox/Frontends/VirtualBox/src/settings/UISettingsDialogSpecific.cpp virtualbox-ose-4.0.0-dfsg/src/VBox/Frontends/VirtualBox/src/settings/UISettingsDialogSpecific.cpp
--- virtualbox-ose-4.0.0-dfsg.orig/src/VBox/Frontends/VirtualBox/src/settings/UISettingsDialogSpecific.cpp	2010-12-17 16:14:24.000000000 +0100
+++ virtualbox-ose-4.0.0-dfsg/src/VBox/Frontends/VirtualBox/src/settings/UISettingsDialogSpecific.cpp	2011-01-02 00:03:55.120533579 +0100
@@ -501,6 +501,10 @@
 #endif /* !VBOX_WITH_NETFLT */
             break;
         }
+        case GLSettingsPage_Update:
+        {
+            return false;
+        }
         default:
             break;
     }
--- VirtualBox-4.1.2_OSE/src/VBox/Frontends/VirtualBox/src/selector/VBoxSelectorWnd.cpp.orig	2011-08-15 14:28:35.000000000 +0200
+++ VirtualBox-4.1.2_OSE/src/VBox/Frontends/VirtualBox/src/selector/VBoxSelectorWnd.cpp	2011-08-17 22:48:08.700558251 +0200
@@ -1747,12 +1747,6 @@
     pMenu->addAction(gActionPool->action(UIActionIndex_Simple_Register));
 #endif /* VBOX_WITH_REGISTRATION */
 
-    pMenu->addAction(gActionPool->action(UIActionIndex_Simple_Update));
-
-#ifndef Q_WS_MAC
-    pMenu->addSeparator();
-#endif /* !Q_WS_MAC */
-
     pMenu->addAction(gActionPool->action(UIActionIndex_Simple_About));
 
     /* Configure connections: */
