diff --color -ur VirtualBox-6.1.18.orig/src/VBox/Runtime/r0drv/linux/memobj-r0drv-linux.c VirtualBox-6.1.18/src/VBox/Runtime/r0drv/linux/memobj-r0drv-linux.c
--- VirtualBox-6.1.18.orig/src/VBox/Runtime/r0drv/linux/memobj-r0drv-linux.c	2021-01-07 16:42:09.000000000 +0100
+++ VirtualBox-6.1.18/src/VBox/Runtime/r0drv/linux/memobj-r0drv-linux.c	2021-01-21 23:13:58.808360771 +0100
@@ -536,7 +536,7 @@
  * @param   uAddr               The address to apply the new protection to.
  * @param   pvUser              The opaque user data.
  */
-static DECLCALLBACK(int) rtR0MemObjLinuxApplyPageRange(pte_t *pPte, unsigned long uAddr, void *pvUser)
+static int rtR0MemObjLinuxApplyPageRange(pte_t *pPte, unsigned long uAddr, void *pvUser)
 {
     PCLNXAPPLYPGRANGE pArgs = (PCLNXAPPLYPGRANGE)pvUser;
     PRTR0MEMOBJLNX pMemLnx = pArgs->pMemLnx;
