--- VirtualBox-6.0.6/src/VBox/HostDrivers/linux/export_modules.sh.orig	2019-04-16 12:16:41.000000000 +0200
+++ VirtualBox-6.0.6/src/VBox/HostDrivers/linux/export_modules.sh	2019-05-05 10:20:35.010948761 +0200
@@ -159,6 +159,7 @@
         < $PATH_VBOXDRV/linux/Makefile > $PATH_TMP/vboxdrv/Makefile
 else
     sed -e "s;-DVBOX_WITH_HARDENING;;g" \
+        -e "s;VBOX_WITH_HARDENING;;g" \
         -e "s;-DVBOX_WITH_EFLAGS_AC_SET_IN_VBOXDRV;;g" \
         -e "s;-DIPRT_WITH_EFLAGS_AC_PRESERVING;;g" \
         < $PATH_VBOXDRV/linux/Makefile > $PATH_TMP/vboxdrv/Makefile
@@ -172,7 +173,9 @@
 if [ -n "$VBOX_WITH_HARDENING" ]; then
     cat                                   $PATH_VBOXNET/linux/Makefile > $PATH_TMP/vboxnetflt/Makefile
 else
-    sed -e "s;-DVBOX_WITH_HARDENING;;g" < $PATH_VBOXNET/linux/Makefile > $PATH_TMP/vboxnetflt/Makefile
+    sed -e "s;-DVBOX_WITH_HARDENING;;g" \
+        -e "s;VBOX_WITH_HARDENING;;g" \
+	    < $PATH_VBOXNET/linux/Makefile > $PATH_TMP/vboxnetflt/Makefile
 fi
 
 # vboxnetadp (VirtualBox network adapter kernel module)
@@ -183,7 +186,9 @@
 if [ -n "$VBOX_WITH_HARDENING" ]; then
     cat                                   $PATH_VBOXADP/linux/Makefile > $PATH_TMP/vboxnetadp/Makefile
 else
-    sed -e "s;-DVBOX_WITH_HARDENING;;g" < $PATH_VBOXADP/linux/Makefile > $PATH_TMP/vboxnetadp/Makefile
+    sed -e "s;-DVBOX_WITH_HARDENING;;g" \
+        -e "s;VBOX_WITH_HARDENING;;g" \
+	    < $PATH_VBOXADP/linux/Makefile > $PATH_TMP/vboxnetadp/Makefile
 fi
 
 # vboxpci (VirtualBox host PCI access kernel module)
@@ -194,7 +199,9 @@
 if [ -n "$VBOX_WITH_HARDENING" ]; then
     cat                                   $PATH_VBOXPCI/linux/Makefile > $PATH_TMP/vboxpci/Makefile
 else
-    sed -e "s;-DVBOX_WITH_HARDENING;;g" < $PATH_VBOXPCI/linux/Makefile > $PATH_TMP/vboxpci/Makefile
+    sed -e "s;-DVBOX_WITH_HARDENING;;g" \
+        -e "s;VBOX_WITH_HARDENING;;g" \
+	    < $PATH_VBOXPCI/linux/Makefile > $PATH_TMP/vboxpci/Makefile
 fi
 
 install -D -m 0644 $PATH_LINUX/Makefile $PATH_TMP/Makefile
