--- VirtualBox-6.1.2/src/VBox/HostDrivers/linux/export_modules.sh.orig	2020-02-14 01:33:04.739918751 +0100
+++ VirtualBox-6.1.2/src/VBox/HostDrivers/linux/export_modules.sh	2020-02-14 01:33:35.779769111 +0100
@@ -173,8 +173,8 @@
         < $PATH_VBOXDRV/linux/Makefile > $PATH_TMP/vboxdrv/Makefile
 else
     sed -e "s;VBOX_WITH_HARDENING;;g" \
-        -e "s;VBOX_WITH_EFLAGS_AC_SET_IN_VBOXDRV;;g" \
-        -e "s;IPRT_WITH_EFLAGS_AC_PRESERVING;;g" \
+        -e "s;VBOX_WITHOUT_EFLAGS_AC_SET_IN_VBOXDRV;;g" \
+        -e "s;IPRT_WITHOUT_EFLAGS_AC_PRESERVING;;g" \
         < $PATH_VBOXDRV/linux/Makefile > $PATH_TMP/vboxdrv/Makefile
 fi
 
