--- VirtualBox-3.2.6_OSE/src/VBox/HostDrivers/linux/export_modules~	2010-08-02 17:33:03.000000000 +0300
+++ VirtualBox-3.2.6_OSE/src/VBox/HostDrivers/linux/export_modules	2010-08-02 17:35:29.736477562 +0300
@@ -109,20 +109,6 @@
     sed -e "s;-DVBOX_WITH_HARDENING;;g" < $PATH_VBOXNET/linux/Makefile > $PATH_TMP/vboxnetflt/Makefile
 fi
 
-# vboxnetadp (VirtualBox network adapter kernel module)
-mkdir $PATH_TMP/vboxnetadp || exit 1
-for f in $VBOX_VBOXNETADP_SOURCES; do
-    install -D -m 0644 `echo $f|cut -d'=' -f1` "$PATH_TMP/vboxnetadp/`echo $f|cut -d'>' -f2`"
-done
-sed -e "s;_VERSION_;$VBOX_VERSION_STRING;g" < $PATH_LINUX/build_in_tmp > $PATH_TMP/vboxnetadp/build_in_tmp
-chmod 0755 $PATH_TMP/vboxnetadp/build_in_tmp
-sed -e "s;_VERSION_;$VBOX_VERSION_STRING;g" < $PATH_VBOXADP/linux/dkms.conf > $PATH_TMP/vboxnetadp/dkms.conf
-if [ -n "$VBOX_WITH_HARDENING" ]; then
-    cat                                   $PATH_VBOXADP/linux/Makefile > $PATH_TMP/vboxnetadp/Makefile
-else
-    sed -e "s;-DVBOX_WITH_HARDENING;;g" < $PATH_VBOXADP/linux/Makefile > $PATH_TMP/vboxnetadp/Makefile
-fi
-
 # vboxnetadp (VirtualBox host-only network kernel module)
 mkdir $PATH_TMP/vboxnetadp || exit 1
 for f in $VBOX_VBOXNETADP_SOURCES; do
