--- VirtualBox-6.1.46/src/VBox/HostDrivers/VBoxNetFlt/linux/VBoxNetFlt-linux.c~	2023-08-18 13:23:19.899777134 +0200
+++ VirtualBox-6.1.46/src/VBox/HostDrivers/VBoxNetFlt/linux/VBoxNetFlt-linux.c	2023-08-18 13:29:51.703087517 +0200
@@ -37,9 +37,9 @@
 #if RTLNX_VER_MIN(2,6,24)
 # include <linux/nsproxy.h>
 #endif
-#if RTLNX_VER_MIN(6,5,0)
+#if RTLNX_VER_MIN(6,4,10)
 # include <net/gso.h>
 #endif
 #include <linux/netdevice.h>
 #if RTLNX_VER_MAX(2,6,29) || RTLNX_VER_MIN(5,11,0)
 # include <linux/ethtool.h>
