--- VirtualBox-6.1.18/src/VBox/HostDrivers/VBoxNetFlt/linux/VBoxNetFlt-linux.c.orig	2021-01-07 16:41:28.000000000 +0100
+++ VirtualBox-6.1.18/src/VBox/HostDrivers/VBoxNetFlt/linux/VBoxNetFlt-linux.c	2021-02-23 19:24:57.308526670 +0100
@@ -39,6 +39,9 @@
 #endif
 #include <linux/netdevice.h>
 #include <linux/etherdevice.h>
+#if RTLNX_VER_MIN(5,11,0)
+#include <linux/ethtool.h>
+#endif
 #include <linux/rtnetlink.h>
 #include <linux/miscdevice.h>
 #include <linux/inetdevice.h>
