--- VirtualBox-6.1.34/kernel/HostDrivers/vboxnetflt/linux/VBoxNetFlt-linux.c~	2022-05-25 17:21:29.000000000 +0200
+++ VirtualBox-6.1.34/kernel/HostDrivers/vboxnetflt/linux/VBoxNetFlt-linux.c	2022-05-26 06:36:50.085761576 +0200
@@ -2310,8 +2310,13 @@
             {
                 vboxNetFltDumpPacket(pSG, true, "host", (fDst & INTNETTRUNKDIR_WIRE) ? 0 : 1);
                 Log6(("vboxNetFltPortOsXmit: pBuf->cb dump:\n%.*Rhxd\n", sizeof(pBuf->cb), pBuf->cb));
+#if RTLNX_VER_MIN(5,18,0)
+                Log6(("vboxNetFltPortOsXmit: netif_rx(%p)\n", pBuf));
+                err = netif_rx(pBuf);
+#else
                 Log6(("vboxNetFltPortOsXmit: netif_rx_ni(%p)\n", pBuf));
                 err = netif_rx_ni(pBuf);
+#endif
                 if (err)
                     rc = RTErrConvertFromErrno(err);
             }
