diff -ur VirtualBox-6.1.18/kernel/GuestDrivers/vboxguest/r0drv/linux/memobj-r0drv-linux.c VirtualBox-6.1.18-256/kernel/GuestDrivers/vboxguest/r0drv/linux/memobj-r0drv-linux.c
--- VirtualBox-6.1.18/kernel/GuestDrivers/vboxguest/r0drv/linux/memobj-r0drv-linux.c	2021-02-06 22:09:20.000000000 +0100
+++ VirtualBox-6.1.18-256/kernel/GuestDrivers/vboxguest/r0drv/linux/memobj-r0drv-linux.c	2021-02-06 22:08:19.038693668 +0100
@@ -1174,7 +1174,7 @@
 
 /* openSUSE Leap 42.3 detection :-/ */
 #if RTLNX_VER_RANGE(4,4,0,  4,6,0) && defined(FAULT_FLAG_REMOTE)
-# define GET_USER_PAGES_API     KERNEL_VERSION(4, 10, 0) /* no typo! */
+# define GET_USER_PAGES_API     KERNEL_VERSION(4, 11, 0) /* no typo! */
 #else
 # define GET_USER_PAGES_API     LINUX_VERSION_CODE
 #endif
@@ -1252,7 +1252,7 @@
 # endif
                                 &pMemLnx->apPages[0],   /* Page array. */
                                 papVMAs                 /* vmas */
-# if GET_USER_PAGES_API >= KERNEL_VERSION(4, 10, 0)
+# if GET_USER_PAGES_API >= KERNEL_VERSION(4, 11, 0)
                                 , NULL                  /* locked */
 # endif
                                 );
diff -ur VirtualBox-6.1.18/kernel/GuestDrivers/vboxguest/r0drv/linux/mpnotification-r0drv-linux.c VirtualBox-6.1.18-256/kernel/GuestDrivers/vboxguest/r0drv/linux/mpnotification-r0drv-linux.c
--- VirtualBox-6.1.18/kernel/GuestDrivers/vboxguest/r0drv/linux/mpnotification-r0drv-linux.c	2021-02-06 22:09:20.000000000 +0100
+++ VirtualBox-6.1.18-256/kernel/GuestDrivers/vboxguest/r0drv/linux/mpnotification-r0drv-linux.c	2021-02-06 21:54:24.869047375 +0100
@@ -37,7 +37,7 @@
 #include <iprt/thread.h>
 #include "r0drv/mp-r0drv.h"
 
-#if RTLNX_VER_MIN(4,10,0)
+#if RTLNX_VER_MIN(4,11,0)
 
 static enum cpuhp_state g_rtR0MpOnline;
 
diff -ur VirtualBox-6.1.18/kernel/GuestDrivers/vboxsf/lnkops.c VirtualBox-6.1.18-256/kernel/GuestDrivers/vboxsf/lnkops.c
--- VirtualBox-6.1.18/kernel/GuestDrivers/vboxsf/lnkops.c	2021-02-06 22:09:20.000000000 +0100
+++ VirtualBox-6.1.18-256/kernel/GuestDrivers/vboxsf/lnkops.c	2021-02-06 21:54:24.869047375 +0100
@@ -286,7 +286,7 @@
  * Symlink inode operations.
  */
 struct inode_operations vbsf_lnk_iops = {
-#if RTLNX_VER_MAX(4,10,0)
+#if RTLNX_VER_MAX(4,11,0)
 # if RTLNX_VER_MIN(2,6,8)
     .readlink    = generic_readlink,
 # else
diff -ur VirtualBox-6.1.18/kernel/GuestDrivers/vboxsf/regops.c VirtualBox-6.1.18-256/kernel/GuestDrivers/vboxsf/regops.c
--- VirtualBox-6.1.18/kernel/GuestDrivers/vboxsf/regops.c	2021-02-06 22:09:20.000000000 +0100
+++ VirtualBox-6.1.18-256/kernel/GuestDrivers/vboxsf/regops.c	2021-02-06 21:54:24.869047375 +0100
@@ -3335,7 +3335,7 @@
 static int vbsf_vmlog_fault(struct vm_area_struct *vma, struct vm_fault *vmf)
 {
     int rc;
-#  if RTLNX_VER_MIN(4,10,0)
+#  if RTLNX_VER_MIN(4,11,0)
     SFLOGFLOW(("vbsf_vmlog_fault: vma=%p vmf=%p flags=%#x addr=%p\n", vma, vmf, vmf->flags, vmf->address));
 #  else
     SFLOGFLOW(("vbsf_vmlog_fault: vma=%p vmf=%p flags=%#x addr=%p\n", vma, vmf, vmf->flags, vmf->virtual_address));
@@ -3384,7 +3384,7 @@
 static int vbsf_vmlog_page_mkwrite(struct vm_area_struct *vma, struct vm_fault *vmf)
 {
     int rc;
-#  if RTLNX_VER_MIN(4,10,0)
+#  if RTLNX_VER_MIN(4,11,0)
     SFLOGFLOW(("vbsf_vmlog_page_mkwrite: vma=%p vmf=%p flags=%#x addr=%p\n", vma, vmf, vmf->flags, vmf->address));
 #  else
     SFLOGFLOW(("vbsf_vmlog_page_mkwrite: vma=%p vmf=%p flags=%#x addr=%p\n", vma, vmf, vmf->flags, vmf->virtual_address));
@@ -3406,7 +3406,7 @@
 
 
 /* Special page fault callback for mapping pages: */
-# if RTLNX_VER_MIN(4,10,0)
+# if RTLNX_VER_MIN(4,11,0)
 static void vbsf_vmlog_map_pages(struct vm_fault *vmf, pgoff_t start, pgoff_t end)
 {
     SFLOGFLOW(("vbsf_vmlog_map_pages: vmf=%p (flags=%#x addr=%p) start=%p end=%p\n", vmf, vmf->flags, vmf->address, start, end));
diff -ur VirtualBox-6.1.18/kernel/GuestDrivers/vboxsf/vfsmod.c VirtualBox-6.1.18-256/kernel/GuestDrivers/vboxsf/vfsmod.c
--- VirtualBox-6.1.18/kernel/GuestDrivers/vboxsf/vfsmod.c	2021-02-06 22:09:20.000000000 +0100
+++ VirtualBox-6.1.18-256/kernel/GuestDrivers/vboxsf/vfsmod.c	2021-02-06 21:54:24.869047375 +0100
@@ -443,7 +443,7 @@
                  trouches it after this point (we may screw up something).  */
 # if RTLNX_VER_RANGE(4,0,0,  4,2,0)
     sb->s_bdi = pSuperInfo->bdi_org; /* (noop_backing_dev_info is not exported) */
-# elif RTLNX_VER_RANGE(2,6,34,  4,10,0)
+# elif RTLNX_VER_RANGE(2,6,34,  4,11,0)
     sb->s_bdi = &noop_backing_dev_info;
 # endif
 #endif
diff -ur VirtualBox-6.1.18/kernel/GuestDrivers/vboxvideo/vbox_ttm.c VirtualBox-6.1.18-256/kernel/GuestDrivers/vboxvideo/vbox_ttm.c
--- VirtualBox-6.1.18/kernel/GuestDrivers/vboxvideo/vbox_ttm.c	2021-02-06 22:09:21.000000000 +0100
+++ VirtualBox-6.1.18-256/kernel/GuestDrivers/vboxvideo/vbox_ttm.c	2021-02-06 21:54:24.872380697 +0100
@@ -314,7 +314,7 @@
 #if RTLNX_VER_MAX(5,10,0)
 	.init_mem_type = vbox_bo_init_mem_type,
 #endif
-#if RTLNX_VER_MIN(4,10,0) || RTLNX_RHEL_MAJ_PREREQ(7,4)
+#if RTLNX_VER_MIN(4,11,0) || RTLNX_RHEL_MAJ_PREREQ(7,4)
 	.eviction_valuable = ttm_bo_eviction_valuable,
 #endif
 	.evict_flags = vbox_bo_evict_flags,
diff -ur VirtualBox-6.1.18/kernel/HostDrivers/vboxdrv/r0drv/linux/memobj-r0drv-linux.c VirtualBox-6.1.18-256/kernel/HostDrivers/vboxdrv/r0drv/linux/memobj-r0drv-linux.c
--- VirtualBox-6.1.18/kernel/HostDrivers/vboxdrv/r0drv/linux/memobj-r0drv-linux.c	2021-02-06 22:09:22.000000000 +0100
+++ VirtualBox-6.1.18-256/kernel/HostDrivers/vboxdrv/r0drv/linux/memobj-r0drv-linux.c	2021-02-06 22:08:40.515390291 +0100
@@ -1174,7 +1174,7 @@
 
 /* openSUSE Leap 42.3 detection :-/ */
 #if RTLNX_VER_RANGE(4,4,0,  4,6,0) && defined(FAULT_FLAG_REMOTE)
-# define GET_USER_PAGES_API     KERNEL_VERSION(4, 10, 0) /* no typo! */
+# define GET_USER_PAGES_API     KERNEL_VERSION(4, 11, 0) /* no typo! */
 #else
 # define GET_USER_PAGES_API     LINUX_VERSION_CODE
 #endif
@@ -1252,7 +1252,7 @@
 # endif
                                 &pMemLnx->apPages[0],   /* Page array. */
                                 papVMAs                 /* vmas */
-# if GET_USER_PAGES_API >= KERNEL_VERSION(4, 10, 0)
+# if GET_USER_PAGES_API >= KERNEL_VERSION(4, 11, 0)
                                 , NULL                  /* locked */
 # endif
                                 );
diff -ur VirtualBox-6.1.18/kernel/HostDrivers/vboxdrv/r0drv/linux/mpnotification-r0drv-linux.c VirtualBox-6.1.18-256/kernel/HostDrivers/vboxdrv/r0drv/linux/mpnotification-r0drv-linux.c
--- VirtualBox-6.1.18/kernel/HostDrivers/vboxdrv/r0drv/linux/mpnotification-r0drv-linux.c	2021-02-06 22:09:22.000000000 +0100
+++ VirtualBox-6.1.18-256/kernel/HostDrivers/vboxdrv/r0drv/linux/mpnotification-r0drv-linux.c	2021-02-06 21:54:24.872380697 +0100
@@ -37,7 +37,7 @@
 #include <iprt/thread.h>
 #include "r0drv/mp-r0drv.h"
 
-#if RTLNX_VER_MIN(4,10,0)
+#if RTLNX_VER_MIN(4,11,0)
 
 static enum cpuhp_state g_rtR0MpOnline;
 
diff -ur VirtualBox-6.1.18/src/VBox/Additions/linux/drm/vbox_ttm.c VirtualBox-6.1.18-256/src/VBox/Additions/linux/drm/vbox_ttm.c
--- VirtualBox-6.1.18/src/VBox/Additions/linux/drm/vbox_ttm.c	2021-01-07 16:34:24.000000000 +0100
+++ VirtualBox-6.1.18-256/src/VBox/Additions/linux/drm/vbox_ttm.c	2021-02-06 21:54:24.869047375 +0100
@@ -319,7 +319,7 @@
 #if RTLNX_VER_MAX(5,10,0)
 	.init_mem_type = vbox_bo_init_mem_type,
 #endif
-#if RTLNX_VER_MIN(4,10,0) || RTLNX_RHEL_MAJ_PREREQ(7,4)
+#if RTLNX_VER_MIN(4,11,0) || RTLNX_RHEL_MAJ_PREREQ(7,4)
 	.eviction_valuable = ttm_bo_eviction_valuable,
 #endif
 	.evict_flags = vbox_bo_evict_flags,
diff -ur VirtualBox-6.1.18/src/VBox/Additions/linux/sharedfolders/lnkops.c VirtualBox-6.1.18-256/src/VBox/Additions/linux/sharedfolders/lnkops.c
--- VirtualBox-6.1.18/src/VBox/Additions/linux/sharedfolders/lnkops.c	2021-01-07 16:34:26.000000000 +0100
+++ VirtualBox-6.1.18-256/src/VBox/Additions/linux/sharedfolders/lnkops.c	2021-02-06 21:54:24.869047375 +0100
@@ -286,7 +286,7 @@
  * Symlink inode operations.
  */
 struct inode_operations vbsf_lnk_iops = {
-#if RTLNX_VER_MAX(4,10,0)
+#if RTLNX_VER_MAX(4,11,0)
 # if RTLNX_VER_MIN(2,6,8)
     .readlink    = generic_readlink,
 # else
diff -ur VirtualBox-6.1.18/src/VBox/Additions/linux/sharedfolders/regops.c VirtualBox-6.1.18-256/src/VBox/Additions/linux/sharedfolders/regops.c
--- VirtualBox-6.1.18/src/VBox/Additions/linux/sharedfolders/regops.c	2021-01-07 16:34:26.000000000 +0100
+++ VirtualBox-6.1.18-256/src/VBox/Additions/linux/sharedfolders/regops.c	2021-02-06 21:54:24.869047375 +0100
@@ -3335,7 +3335,7 @@
 static int vbsf_vmlog_fault(struct vm_area_struct *vma, struct vm_fault *vmf)
 {
     int rc;
-#  if RTLNX_VER_MIN(4,10,0)
+#  if RTLNX_VER_MIN(4,11,0)
     SFLOGFLOW(("vbsf_vmlog_fault: vma=%p vmf=%p flags=%#x addr=%p\n", vma, vmf, vmf->flags, vmf->address));
 #  else
     SFLOGFLOW(("vbsf_vmlog_fault: vma=%p vmf=%p flags=%#x addr=%p\n", vma, vmf, vmf->flags, vmf->virtual_address));
@@ -3384,7 +3384,7 @@
 static int vbsf_vmlog_page_mkwrite(struct vm_area_struct *vma, struct vm_fault *vmf)
 {
     int rc;
-#  if RTLNX_VER_MIN(4,10,0)
+#  if RTLNX_VER_MIN(4,11,0)
     SFLOGFLOW(("vbsf_vmlog_page_mkwrite: vma=%p vmf=%p flags=%#x addr=%p\n", vma, vmf, vmf->flags, vmf->address));
 #  else
     SFLOGFLOW(("vbsf_vmlog_page_mkwrite: vma=%p vmf=%p flags=%#x addr=%p\n", vma, vmf, vmf->flags, vmf->virtual_address));
@@ -3406,7 +3406,7 @@
 
 
 /* Special page fault callback for mapping pages: */
-# if RTLNX_VER_MIN(4,10,0)
+# if RTLNX_VER_MIN(4,11,0)
 static void vbsf_vmlog_map_pages(struct vm_fault *vmf, pgoff_t start, pgoff_t end)
 {
     SFLOGFLOW(("vbsf_vmlog_map_pages: vmf=%p (flags=%#x addr=%p) start=%p end=%p\n", vmf, vmf->flags, vmf->address, start, end));
diff -ur VirtualBox-6.1.18/src/VBox/Additions/linux/sharedfolders/vfsmod.c VirtualBox-6.1.18-256/src/VBox/Additions/linux/sharedfolders/vfsmod.c
--- VirtualBox-6.1.18/src/VBox/Additions/linux/sharedfolders/vfsmod.c	2021-01-07 16:34:26.000000000 +0100
+++ VirtualBox-6.1.18-256/src/VBox/Additions/linux/sharedfolders/vfsmod.c	2021-02-06 21:54:24.869047375 +0100
@@ -443,7 +443,7 @@
                  trouches it after this point (we may screw up something).  */
 # if RTLNX_VER_RANGE(4,0,0,  4,2,0)
     sb->s_bdi = pSuperInfo->bdi_org; /* (noop_backing_dev_info is not exported) */
-# elif RTLNX_VER_RANGE(2,6,34,  4,10,0)
+# elif RTLNX_VER_RANGE(2,6,34,  4,11,0)
     sb->s_bdi = &noop_backing_dev_info;
 # endif
 #endif
diff -ur VirtualBox-6.1.18/src/VBox/Runtime/r0drv/linux/memobj-r0drv-linux.c VirtualBox-6.1.18-256/src/VBox/Runtime/r0drv/linux/memobj-r0drv-linux.c
--- VirtualBox-6.1.18/src/VBox/Runtime/r0drv/linux/memobj-r0drv-linux.c	2021-02-06 22:09:20.335448500 +0100
+++ VirtualBox-6.1.18-256/src/VBox/Runtime/r0drv/linux/memobj-r0drv-linux.c	2021-02-06 22:07:34.318634767 +0100
@@ -1174,7 +1174,7 @@
 
 /* openSUSE Leap 42.3 detection :-/ */
 #if RTLNX_VER_RANGE(4,4,0,  4,6,0) && defined(FAULT_FLAG_REMOTE)
-# define GET_USER_PAGES_API     KERNEL_VERSION(4, 10, 0) /* no typo! */
+# define GET_USER_PAGES_API     KERNEL_VERSION(4, 11, 0) /* no typo! */
 #else
 # define GET_USER_PAGES_API     LINUX_VERSION_CODE
 #endif
@@ -1252,7 +1252,7 @@
 # endif
                                 &pMemLnx->apPages[0],   /* Page array. */
                                 papVMAs                 /* vmas */
-# if GET_USER_PAGES_API >= KERNEL_VERSION(4, 10, 0)
+# if GET_USER_PAGES_API >= KERNEL_VERSION(4, 11, 0)
                                 , NULL                  /* locked */
 # endif
                                 );
diff -ur VirtualBox-6.1.18/src/VBox/Runtime/r0drv/linux/mpnotification-r0drv-linux.c VirtualBox-6.1.18-256/src/VBox/Runtime/r0drv/linux/mpnotification-r0drv-linux.c
--- VirtualBox-6.1.18/src/VBox/Runtime/r0drv/linux/mpnotification-r0drv-linux.c	2021-01-07 16:42:09.000000000 +0100
+++ VirtualBox-6.1.18-256/src/VBox/Runtime/r0drv/linux/mpnotification-r0drv-linux.c	2021-02-06 21:54:24.869047375 +0100
@@ -37,7 +37,7 @@
 #include <iprt/thread.h>
 #include "r0drv/mp-r0drv.h"
 
-#if RTLNX_VER_MIN(4,10,0)
+#if RTLNX_VER_MIN(4,11,0)
 
 static enum cpuhp_state g_rtR0MpOnline;
 
