--- VirtualBox-4.2.12/./src/VBox/Main/xml/Settings.cpp~	2013-04-12 13:38:26.000000000 +0300
+++ VirtualBox-4.2.12/./src/VBox/Main/xml/Settings.cpp	2013-05-13 15:49:30.035999373 +0300
@@ -3408,6 +3408,7 @@
     { "xandros", "Xandros" },
     { "freebsd", "FreeBSD" },
     { "openbsd", "OpenBSD" },
+    { "pld", "PLD" },
     { "netbsd", "NetBSD" },
     { "netware", "Netware" },
     { "solaris", "Solaris" },
--- VirtualBox-4.2.12/./src/VBox/Frontends/VirtualBox/VirtualBox1.qrc~	2013-04-12 13:37:50.000000000 +0300
+++ VirtualBox-4.2.12/./src/VBox/Frontends/VirtualBox/VirtualBox1.qrc	2013-05-13 15:51:14.323890458 +0300
@@ -26,6 +26,8 @@
         <file alias="os_netware.png">images/os_netware.png</file>
         <file alias="os_openbsd.png">images/os_openbsd.png</file>
         <file alias="os_openbsd_64.png">images/os_openbsd_64.png</file>
+        <file alias="os_pld.png">images/os_pld.png</file>
+        <file alias="os_pld_64.png">images/os_pld_64.png</file>
         <file alias="os_opensuse.png">images/os_opensuse.png</file>
         <file alias="os_opensuse_64.png">images/os_opensuse_64.png</file>
         <file alias="os_oracle.png">images/os_oracle.png</file>
--- VirtualBox-5.1.2/src/VBox/Main/src-all/Global.cpp.orig	2016-07-29 19:25:16.731328887 +0200
+++ VirtualBox-5.1.2/src/VBox/Main/src-all/Global.cpp	2016-07-29 19:31:59.252427682 +0200
@@ -321,6 +321,17 @@
        512,  16,  8 * _1G64, NetworkAdapterType_I82540EM, 0, StorageControllerType_PIIX4, StorageBus_IDE,
         StorageControllerType_PIIX4, StorageBus_IDE, ChipsetType_PIIX3, AudioControllerType_AC97, AudioCodecType_STAC9700  },
 
+    { "Linux",   "Linux",             "PLD",            "PLD Linux (32-bit)",
+      VBOXOSTYPE_Linux26,         VBOXOSHINT_RTCUTC | VBOXOSHINT_USBTABLET | VBOXOSHINT_X2APIC,
+       512,  16,  8 * _1G64, NetworkAdapterType_I82540EM, 0, StorageControllerType_PIIX4, StorageBus_IDE,
+        StorageControllerType_IntelAhci, StorageBus_SATA, ChipsetType_PIIX3, AudioControllerType_AC97, AudioCodecType_AD1980  },
+
+    { "Linux",   "Linux",             "PLD_64",         "PLD Linux (64-bit)",
+      VBOXOSTYPE_Linux26_x64,     VBOXOSHINT_64BIT | VBOXOSHINT_HWVIRTEX | VBOXOSHINT_IOAPIC | VBOXOSHINT_RTCUTC 
+                                | VBOXOSHINT_USBTABLET | VBOXOSHINT_X2APIC,
+      1024,  16,  8 * _1G64, NetworkAdapterType_I82540EM, 0, StorageControllerType_PIIX4, StorageBus_IDE,
+        StorageControllerType_IntelAhci, StorageBus_SATA, ChipsetType_PIIX3, AudioControllerType_AC97, AudioCodecType_AD1980  },
+
     { "Solaris", "Solaris",           "Solaris",            "Oracle Solaris 10 5/09 and earlier (32-bit)",
       VBOXOSTYPE_Solaris,         VBOXOSHINT_NONE,
        768,  16, 16 * _1G64, NetworkAdapterType_I82540EM, 0, StorageControllerType_PIIX4, StorageBus_IDE,
--- VirtualBox-4.3.8/src/VBox/Main/src-server/ApplianceImpl.cpp~	2014-02-25 19:09:15.000000000 +0200
+++ VirtualBox-4.3.8/src/VBox/Main/src-server/ApplianceImpl.cpp	2014-02-28 09:52:48.663191185 +0200
@@ -141,6 +141,8 @@
     { ovf::CIMOSType_CIMOS_Linux_2_6_x_64,                       VBOXOSTYPE_Xandros_x64 },
     { ovf::CIMOSType_CIMOS_Solaris,                              VBOXOSTYPE_OpenSolaris },
     { ovf::CIMOSType_CIMOS_Solaris_64,                           VBOXOSTYPE_OpenSolaris_x64 },
+    { ovf::CIMOSType_CIMOS_Linux_2_6_x,                          VBOXOSTYPE_PLD },
+    { ovf::CIMOSType_CIMOS_Linux_2_6_x_64,                       VBOXOSTYPE_PLD_x64 },
 
     // types added with CIM 2.25.0 follow:
     { ovf::CIMOSType_CIMOS_WindowsServer2008R2,                  VBOXOSTYPE_Win2k8 },           // duplicate, see above
--- VirtualBox-4.3.8/include/VBox/ostypes.h~	2014-02-25 18:59:48.000000000 +0200
+++ VirtualBox-4.3.8/include/VBox/ostypes.h	2014-02-28 09:56:34.440314457 +0200
@@ -103,6 +103,8 @@
     VBOXOSTYPE_Xandros_x64      = 0x5D100,
     VBOXOSTYPE_Oracle           = 0x5E000,
     VBOXOSTYPE_Oracle_x64       = 0x5E100,
+    VBOXOSTYPE_PLD              = 0x5F000,
+    VBOXOSTYPE_PLD_x64          = 0x5F100,
     VBOXOSTYPE_FreeBSD          = 0x60000,
     VBOXOSTYPE_FreeBSD_x64      = 0x60100,
     VBOXOSTYPE_OpenBSD          = 0x61000,
--- VirtualBox-4.3.8/src/VBox/Frontends/VirtualBox/src/wizards/newvm/UIWizardNewVMPageBasic1.cpp~	2014-02-25 19:09:00.000000000 +0200
+++ VirtualBox-4.3.8/src/VBox/Frontends/VirtualBox/src/wizards/newvm/UIWizardNewVMPageBasic1.cpp	2014-02-28 09:58:59.493698505 +0200
@@ -133,6 +133,8 @@
     { QRegExp("Xa.*32",                            Qt::CaseInsensitive), "Xandros" },
     { QRegExp("((Or)|(oel)|(ol)).*64",             Qt::CaseInsensitive), "Oracle_64" },
     { QRegExp("((Or)|(oel)|(ol)).*32",             Qt::CaseInsensitive), "Oracle" },
+    { QRegExp("PLD.*64",                           Qt::CaseInsensitive), "PLD_64" },
+    { QRegExp("PLD.*32",                           Qt::CaseInsensitive), "PLD" },
     { QRegExp("Knoppix",                           Qt::CaseInsensitive), "Linux26" },
     { QRegExp("Dsl",                               Qt::CaseInsensitive), "Linux24" },
     { QRegExp("((Lin)|(lnx)).*2.?2",               Qt::CaseInsensitive), "Linux22" },
--- VirtualBox-5.0.0/src/VBox/./Frontends/VirtualBox/src/globals/UIIconPool.cpp.orig	2015-07-09 20:00:01.339965485 +0200
+++ VirtualBox-5.0.0/src/VBox/./Frontends/VirtualBox/src/globals/UIIconPool.cpp	2015-07-09 20:00:40.670058835 +0200
@@ -332,6 +332,8 @@
     m_guestOSTypeIconNames.insert("FreeBSD_64",      ":/os_freebsd_64.png");
     m_guestOSTypeIconNames.insert("OpenBSD",         ":/os_openbsd.png");
     m_guestOSTypeIconNames.insert("OpenBSD_64",      ":/os_openbsd_64.png");
+    m_guestOSTypeIconNames.insert("PLD",             ":/os_pld.png");
+    m_guestOSTypeIconNames.insert("PLD_64",          ":/os_pld_64.png");
     m_guestOSTypeIconNames.insert("NetBSD",          ":/os_netbsd.png");
     m_guestOSTypeIconNames.insert("NetBSD_64",       ":/os_netbsd_64.png");
     m_guestOSTypeIconNames.insert("Solaris",         ":/os_solaris.png");

