--- VirtualBox-7.0.10/configure~	2023-09-06 15:45:03.000000000 +0200
+++ VirtualBox-7.0.10/configure	2023-09-06 15:48:44.766964239 +0200
@@ -2173,7 +2173,7 @@
 }
 EOF
   found=
-  SUPPYTHONLIBS="python2.7 python2.6 python3.1 python3.2 python3.3 python3.4 python3.4m python3.5 python3.5m python3.6 python3.6m python3.7 python3.7m python3.8 python3.9 python3.10 python3.11 python3.12"
+  SUPPYTHONLIBS="python3.1 python3.2 python3.3 python3.4 python3.4m python3.5 python3.5m python3.6 python3.6m python3.7 python3.7m python3.8 python3.9 python3.10 python3.11 python3.12"
   for p in $PYTHONDIR; do
     for d in $SUPPYTHONLIBS; do
       for b in lib/x86_64-linux-gnu lib/i386-linux-gnu lib64 lib/64 lib; do
--- VirtualBox-7.0.10/src/libs/xpcom18a4/python/gen_python_deps.py~	2023-07-12 18:42:54.000000000 +0200
+++ VirtualBox-7.0.10/src/libs/xpcom18a4/python/gen_python_deps.py	2023-09-07 16:25:51.601179500 +0200
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 """
 Copyright (C) 2009-2023 Oracle and/or its affiliates.
@@ -26,7 +26,7 @@
 import os,sys
 from distutils.version import StrictVersion
 
-versions = ["2.6", "2.7", "3.1", "3.2", "3.2m", "3.3", "3.3m", "3.4", "3.4m", "3.5", "3.5m", "3.6", "3.6m", "3.7", "3.7m", "3.8", "3.9", "3.10", "3.11", "3.12" ]
+versions = ["3.1", "3.2", "3.2m", "3.3", "3.3m", "3.4", "3.4m", "3.5", "3.5m", "3.6", "3.6m", "3.7", "3.7m", "3.8", "3.9", "3.10", "3.11", "3.12" ]
 prefixes = ["/usr", "/usr/local", "/opt", "/opt/local"]
 known = {}
 
--- VirtualBox-7.0.10/Config.kmk~	2023-09-08 01:41:50.000000000 +0200
+++ VirtualBox-7.0.10/Config.kmk	2023-09-08 01:51:41.947941370 +0200
@@ -853,7 +853,7 @@
 # unless VBOX_WITH_ONLY_PYTHON_LIMITED_API is overridden).
 VBOX_WITH_MULTIVERSION_PYTHON ?= 1
 # Only build the limited Python extension API version for 3.3 and later.
-VBOX_WITH_ONLY_PYTHON_LIMITED_API = 1
+# VBOX_WITH_ONLY_PYTHON_LIMITED_API = 1
 # Build the Validation Kit.
 VBOX_WITH_VALIDATIONKIT = 1
 # Include unit tests (testcases) on the Validation Kit .ISO
