--- VirtualBox-6.1.26/configure~	2021-09-10 08:38:49.000000000 +0200
+++ VirtualBox-6.1.26/configure	2021-09-10 09:01:11.297646984 +0200
@@ -1564,7 +1564,7 @@
           FLGQT5=`pkg-config Qt5Core --cflags`
           # gcc 4.8 is able to compile with C++11 (see also VBOX_GCC_std in Config.kmk)
           [ $(($cc_maj * 100 + $cc_min)) -ge 408 ] && FLGQT5="$FLGQT5 -std=c++11"
-          INCQT5=`strip_I "$FLGQT5"`
+          INCQT5=`pkg-config Qt5Core --variable=includedir`
           LIBDIR5=`pkg-config Qt5Core --variable=libdir`
           LIBQT5=`pkg-config Qt5Core Qt5Gui --libs`
           LIBQT5="-L$LIBDIR5 $LIBQT5"
