diff -urN VirtualBox-2.0.0.orig/configure VirtualBox-2.0.0/configure
--- VirtualBox-2.0.0.orig/configure	2008-09-04 20:06:22.000000000 +0200
+++ VirtualBox-2.0.0/configure	2008-09-04 20:07:39.000000000 +0200
@@ -1401,29 +1401,7 @@
 }
 
 
-#
-# Check whether static libstdc++ is installed
-#
-check_staticlibstdcxx()
-{
-  test_header "static stc++ library"
-  libstdcxx=`$CXX -print-file-name=libstdc++.a`
-  cat > .tmp_src.cc << EOF
-#include <string>
-
-extern "C" int main(void)
-{
-  std::string s = "test";
-  return 0;
-}
-EOF
-  if test_compile "$libstdcxx" libstdc++ libstdc++; then
-    log_success "found"
-  fi
-}
-
 
-#
 # Check for Linux sources
 #
 check_linux()
@@ -1996,7 +1974,6 @@
 
 # Linux-specific
 if [ "$OS" = "linux" ]; then
-  check_staticlibstdcxx
   if [ $WITH_KMODS -eq 1 ]; then
     check_linux
   else
diff -urN VirtualBox-2.0.0.orig/src/VBox/Additions/x11/xclient/Makefile.kmk VirtualBox-2.0.0/src/VBox/Additions/x11/xclient/Makefile.kmk
--- VirtualBox-2.0.0.orig/src/VBox/Additions/x11/xclient/Makefile.kmk	2008-09-01 17:26:05.000000000 +0200
+++ VirtualBox-2.0.0/src/VBox/Additions/x11/xclient/Makefile.kmk	2008-09-04 20:07:39.000000000 +0200
@@ -74,14 +74,14 @@
 #
 # Actually, this is darn annoying and will *NOT* be tolerated for any new code!
 #
-VBoxClient_LIBPATH  += $(PATH_TARGET)/VBoxClient
-VBoxClient_ORDERDEPS = $(PATH_TARGET)/VBoxClient/libstdc++.a
-VBoxClient_CLEAN     = $(PATH_TARGET)/VBoxClient/libstdc++.a
-$(PATH_TARGET)/VBoxClient/libstdc++.a:
-	$(call MSG_L1,Forcing static libstdc++)
-	$(QUIET)$(MKDIR) -p $(@D)
-	$(QUIET)$(LN_EXT) -sf `$(TOOL_$(VBOX_GCC_TOOL)_CXX) $(TEMPLATE_VBOXGUESTR3CPPEXE_CXXFLAGS.$(KBUILD_TARGET_ARCH)) -print-file-name=libstdc++.a` $@ \
-	     || $(CP_EXT) -f  `$(TOOL_$(VBOX_GCC_TOOL)_CXX) $(TEMPLATE_VBOXGUESTR3CPPEXE_CXXFLAGS.$(KBUILD_TARGET_ARCH)) -print-file-name=libstdc++.a` $@
+#VBoxClient_LIBPATH  += $(PATH_TARGET)/VBoxClient
+#VBoxClient_ORDERDEPS = $(PATH_TARGET)/VBoxClient/libstdc++.a
+#VBoxClient_CLEAN     = $(PATH_TARGET)/VBoxClient/libstdc++.a
+#$(PATH_TARGET)/VBoxClient/libstdc++.a:
+#	$(call MSG_L1,Forcing static libstdc++)
+#	$(QUIET)$(MKDIR) -p $(@D)
+#	$(QUIET)$(LN_EXT) -sf `$(TOOL_$(VBOX_GCC_TOOL)_CXX) $(TEMPLATE_VBOXGUESTR3CPPEXE_CXXFLAGS.$(KBUILD_TARGET_ARCH)) -print-file-name=libstdc++.a` $@ \
+#	     || $(CP_EXT) -f  `$(TOOL_$(VBOX_GCC_TOOL)_CXX) $(TEMPLATE_VBOXGUESTR3CPPEXE_CXXFLAGS.$(KBUILD_TARGET_ARCH)) -print-file-name=libstdc++.a` $@
 
 
 ifdef VBOX_X11_SEAMLESS_GUEST
