diff -urN VirtualBox-2.0.0.orig/src/VBox/Additions/x11/Makefile.kmk VirtualBox-2.0.0/src/VBox/Additions/x11/Makefile.kmk
--- VirtualBox-2.0.0.orig/src/VBox/Additions/x11/Makefile.kmk	2008-09-01 17:26:06.000000000 +0200
+++ VirtualBox-2.0.0/src/VBox/Additions/x11/Makefile.kmk	2008-09-04 20:09:32.000000000 +0200
@@ -28,7 +28,6 @@
 
 # Include sub-makefiles.
 if1of ($(KBUILD_TARGET), freebsd linux netbsd openbsd solaris)
- include $(PATH_SUB_CURRENT)/xclient/Makefile.kmk
  include $(PATH_SUB_CURRENT)/xgraphics/Makefile.kmk
  include $(PATH_SUB_CURRENT)/xmouse/Makefile.kmk
 endif
