--- VirtualBox-4.0.0_OSE/PLD-MODULE-BUILD/HostDrivers/vboxnetflt/linux/VBoxNetFlt-linux.c~	2010-12-29 16:48:42.000000000 +0100
+++ VirtualBox-4.0.0_OSE/PLD-MODULE-BUILD/HostDrivers/vboxnetflt/linux/VBoxNetFlt-linux.c	2010-12-29 16:50:09.328529008 +0100
@@ -51,7 +51,7 @@
 #include "../VBoxNetFltInternal.h"
 
 #ifdef CONFIG_NET_SCHED
-# define VBOXNETFLT_WITH_QDISC /* Comment this out to disable qdisc support */
+//# define VBOXNETFLT_WITH_QDISC /* Comment this out to disable qdisc support */
 # ifdef VBOXNETFLT_WITH_QDISC
 # include <net/pkt_sched.h>
 # endif /* VBOXNETFLT_WITH_QDISC */
