--- VirtualBox-4.0.4_OSE/configure~	2011-03-28 14:11:08.533832064 +0200
+++ VirtualBox-4.0.4_OSE/configure	2011-03-28 14:12:30.980943864 +0200
@@ -409,7 +409,7 @@
       elif [ $cc_maj -lt 3 \
              -o \( $cc_maj -eq 3 -a $cc_min -lt 2 \) \
              -o \( $cc_maj -eq 4 -a $cc_min -lt 1 -a "$OS" != "darwin" \) \
-             -o \( $cc_maj -eq 4 -a $cc_min -gt 5 \) \
+             -o \( $cc_maj -eq 4 -a $cc_min -gt 6 \) \
              -o $cc_maj -gt 4 ]; then
         log_failure "gcc version $cc_ver found, expected gcc 3.x with x>1 or gcc 4.x with 0<x<5"
         fail really
