--- VirtualBox-3.1.6_OSE/configure~	2010-04-28 15:48:34.341923836 +0200
+++ VirtualBox-3.1.6_OSE/configure	2010-04-28 15:49:21.158168248 +0200
@@ -394,9 +394,9 @@
       elif [ $cc_maj -lt 3 \
              -o \( $cc_maj -eq 3 -a $cc_min -lt 2 \) \
              -o \( $cc_maj -eq 4 -a $cc_min -lt 1 -a "$OS" != "darwin" \) \
-             -o \( $cc_maj -eq 4 -a $cc_min -gt 4 \) \
+             -o \( $cc_maj -eq 4 -a $cc_min -gt 5 \) \
              -o $cc_maj -gt 4 ]; then
-        log_failure "gcc version $cc_ver found, expected gcc 3.x with x>1 or gcc 4.x with 0<x<4"
+        log_failure "gcc version $cc_ver found, expected gcc 3.x with x>1 or gcc 4.x with 0<x<5"
         fail really
       else
         log_success "found version $cc_ver"
