diff -urN VirtualBox-1.5.0_OSE.orig/src/VBox/Additions/linux/Makefile.kmk VirtualBox-1.5.0_OSE/src/VBox/Additions/linux/Makefile.kmk
--- VirtualBox-1.5.0_OSE.orig/src/VBox/Additions/linux/Makefile.kmk	2007-08-30 17:16:27.000000000 +0200
+++ VirtualBox-1.5.0_OSE/src/VBox/Additions/linux/Makefile.kmk	2007-10-24 09:39:16.000000000 +0200
@@ -27,8 +27,7 @@
  ifdef VBOX_WITH_X11_ADDITIONS
  SUBDIRS += \
 	xgraphics \
-	xmouse \
-	xclient
+	xmouse
  endif
 endif
 
