------------------------------------------------------------------------
r57658 | vboxsync | 2015-09-09 12:54:30 +0200 (śro, 09 wrz 2015) | 2 lines

VNC: build fix


Index: src/VBox/ExtPacks/VNC/VBoxVNC.cpp
===================================================================
--- src/VBox/ExtPacks/VNC/VBoxVNC.cpp	(revision 57657)
+++ src/VBox/ExtPacks/VNC/VBoxVNC.cpp	(revision 57658)
@@ -103,8 +103,8 @@
     unsigned char *mScreenBuffer;
     unsigned char *mFrameBuffer;
     uint32_t uClients;
-    static DECLCALLBACK(enum rfbNewClientAction) rfbNewClientEvent(rfbClientPtr cl);
-    static DECLCALLBACK(void) vncMouseEvent(int buttonMask, int x, int y, rfbClientPtr cl);
+    static enum rfbNewClientAction rfbNewClientEvent(rfbClientPtr cl);
+    static void vncMouseEvent(int buttonMask, int x, int y, rfbClientPtr cl);
     static void vncKeyboardEvent(rfbBool down, rfbKeySym keySym, rfbClientPtr cl);
     static void clientGoneHook(rfbClientPtr cl);
 

------------------------------------------------------------------------
------------------------------------------------------------------------
r57690 | vboxsync | 2015-09-10 15:23:58 +0200 (czw, 10 wrz 2015) | 2 lines

Additions/vbox-greeter: build fix


Index: src/VBox/Additions/linux/lightdm-greeter/vbox-greeter.cpp
===================================================================
--- src/VBox/Additions/linux/lightdm-greeter/vbox-greeter.cpp	(revision 57689)
+++ src/VBox/Additions/linux/lightdm-greeter/vbox-greeter.cpp	(revision 57690)
@@ -842,7 +842,7 @@
  * @param   enmPhase
  * @param   pfnLog
  */
-static void vboxGreeterLogHeaderFooter(PRTLOGGER pLoggerRelease, RTLOGPHASE enmPhase, PFNRTLOGPHASEMSG pfnLog)
+static DECLCALLBACK(void) vboxGreeterLogHeaderFooter(PRTLOGGER pLoggerRelease, RTLOGPHASE enmPhase, PFNRTLOGPHASEMSG pfnLog)
 {
     /* Some introductory information. */
     static RTTIMESPEC s_TimeSpec;

------------------------------------------------------------------------
