--- VirtualBox-3.1.8_OSE/src/VBox/Additions/linux/export_modules.orig	2010-05-10 08:45:29.000000000 +0200
+++ VirtualBox-3.1.8_OSE/src/VBox/Additions/linux/export_modules	2010-05-16 14:15:16.321158339 +0200
@@ -71,6 +73,15 @@
     install -D -m 0755 `echo $f|cut -d'=' -f1` "$PATH_TMP/vboxsf/`echo $f|cut -d'>' -f2`"
 done
 
+# vboxvideo_drm (VirtualBox guest kernel module for DRI)
+mkdir $PATH_TMP/vboxvideo_drm || exit 1
+for f in $FILES_VBOXVIDEO_DRM_NOBIN; do
+    install -D -m 0644 `echo $f|cut -d'=' -f1` "$PATH_TMP/vboxvideo_drm/`echo $f|cut -d'>' -f2`"
+done
+for f in $FILES_VBOXVIDEO_DRM_BIN; do
+    install -D -m 0755 `echo $f|cut -d'=' -f1` "$PATH_TMP/vboxvideo_drm/`echo $f|cut -d'>' -f2`"
+done
+
 # Only temporary, omit from archive
 rm $PATH_TMP/version-generated.h
 
