--- src/VBox/HostDrivers/VBoxNetFlt/Makefile.kmk.orig	2008-12-19 11:09:46.000000000 +0100
+++ src/VBox/HostDrivers/VBoxNetFlt/Makefile.kmk	2008-12-19 12:47:44.000000000 +0100
@@ -221,6 +221,7 @@
 # vboxnetflt(.ko/.o/) - The lower case driver.
 # Note! On Solaris the name has to be <= 8 chars long.
 #
+ ifdef VBOX_WITH_VBOXDRV
 SYSMODS += vboxnetflt
 vboxnetflt_TEMPLATE      = VBOXR0DRV
 vboxnetflt_NAME.solaris  = vboxflt
@@ -246,7 +247,7 @@
 vboxnetflt_SOURCES.linux = linux/VBoxNetFlt-linux.c
 vboxnetflt_SOURCES.solaris = solaris/VBoxNetFlt-solaris.c
 vboxnetflt_SOURCES = VBoxNetFlt.c
-
+ endif
 endif
 
 ifeq ($(KBUILD_TARGET),linux)
