--- VirtualBox-1.5.4_OSE/src/VBox/Additions/linux/xclient/Makefile.kmk	2007-12-30 17:02:00.000000000 +0000
+++ VirtualBox-1.5.4_OSE/src/VBox/Additions/linux/xclient/Makefile.kmk	2007-12-30 17:05:37.000000000 +0000
@@ -32,10 +32,10 @@
 vboxadd-xclient_DEFS     = _GNU_SOURCE VBOX_HGCM
 
 # link agaginst libstdc++.a (http://www.trilithium.com/johan/2005/06/static-libstdc/)
-$(PATH_TARGET)/vboxadd-xclient/vboxadd-xclient: $(PATH_TARGET)/vboxadd-xclient/libstdc++.a
+#$(PATH_TARGET)/vboxadd-xclient/vboxadd-xclient: $(PATH_TARGET)/vboxadd-xclient/libstdc++.a
 
-$(PATH_TARGET)/vboxadd-xclient/libstdc++.a:
-	$(call MSG_L1,Forcing static libstdc++)
-	$(QUIET)ln -sf `$(TOOL_$(VBOX_GCC32_TOOL)_CXX) -print-file-name=libstdc++.a` $@ || cp `$(TOOL_$(VBOX_GCC32_TOOL)_CXX) -print-file-name=libstdc++.a` $@
+#$(PATH_TARGET)/vboxadd-xclient/libstdc++.a:
+#	$(call MSG_L1,Forcing static libstdc++)
+#	$(QUIET)ln -sf `$(TOOL_$(VBOX_GCC32_TOOL)_CXX) -print-file-name=libstdc++.a` $@ || cp `$(TOOL_$(VBOX_GCC32_TOOL)_CXX) -print-file-name=libstdc++.a` $@
 
 include	$(PATH_KBUILD)/footer.kmk
