--- VirtualBox-4.1.0_OSE/src/VBox/HostDrivers/VBoxPci/Makefile.kmk.orig	2011-04-28 10:01:04.000000000 +0200
+++ VirtualBox-4.1.0_OSE/src/VBox/HostDrivers/VBoxPci/Makefile.kmk	2011-07-19 20:51:04.149020310 +0200
@@ -5,6 +5,7 @@
 
 if1of ($(KBUILD_TARGET), linux)
 
+ ifdef VBOX_WITH_VBOXDRV
 SYSMODS += VBoxPci
 VBoxPci_TEMPLATE         = VBOXR0DRV
 VBoxPci_INST = $(INST_VBOXPCI)$(if $(eq $(KBUILD_TARGET),darwin),Contents/MacOS/)
@@ -22,7 +23,7 @@
 VBoxPci_SOURCES          =
 VBoxPci_LIBS            += \
 	$(PATH_LIB)/SUPR0IdcClient$(VBOX_SUFF_LIB)
-
+ endif
 endif
 
 
