--- configure~	2008-08-04 23:49:45.338189943 +0200
+++ configure	2008-08-05 08:45:58.844848232 +0200
@@ -334,35 +334,6 @@
         fail really
       elif [ $cc_maj -eq 4 -a $cc_min -eq 0 -a "OS" = "darwin" ]; then
         log_success "found version $cc_ver"
-      elif [ $cc_maj -eq 4 -a $cc_min -eq 3 ]; then
-        if [ -z "$CC_COMPAT" ]; then
-          log ""
-          log "  ** There are known problems with gcc version 4.3 when compiling the recompiler"
-          log "  ** stuff. You need to specify a compatibility compiler with version < 4.3. Look"
-          log "  ** for a package compat-gcc-34 on Fedora systems or something similar on other"
-          log "  ** distributions and call configure with parameter --with-gcc-compat=gcc34."
-          fail really
-        fi
-        if check_avail "$CC_COMPAT" CC_COMPAT really; then
-          cc_compat_ver=`$CC_COMPAT -dumpversion` 2>/dev/null
-          if [ $? -ne 0 ]; then
-            log_failure "cannot execute '$CC_COMPAT -dumpversion'"
-            fail really
-          fi
-          cc_compat_maj=`echo $cc_compat_ver|cut -d. -f1`
-          cc_compat_min=`echo $cc_compat_ver|cut -d. -f2`
-          if [ $cc_compat_maj -lt 3 \
-               -o \( $cc_compat_maj -eq 3 -a $cc_compat_min -lt 2 \) \
-               -o \( $cc_compat_maj -eq 4 -a $cc_compat_min -lt 1 \) \
-               -o \( $cc_compat_maj -eq 4 -a $cc_compat_min -gt 2 \) \
-               -o $cc_compat_maj -gt 4 ]; then
-            log ""
-            log "  ** Version $cc_compat_ver of the compatibility gcc found. Expected gcc 3.x with x>1"
-            log "  ** or gcc 4.x with 0<x<3"
-            fail really
-          fi
-          cnf_append "VBOX_RECOMPILER_OP_GCC" "$CC_COMPAT"
-        fi
       # gcc-4.0 is allowed for Darwin only
       elif [ $cc_maj -lt 3 \
              -o \( $cc_maj -eq 3 -a $cc_min -lt 2 \) \
--- src/recompiler/Makefile.kmk~	2008-08-05 10:48:06.537032700 +0200
+++ src/recompiler/Makefile.kmk	2008-08-05 10:49:59.933687528 +0200
@@ -353,9 +353,9 @@
 
 
 # The command sans -o op.S.tmp.
-COMPILE_OP_CMDS = $(VBOX_RECOMPILER_OP_GCC) \
+COMPILE_OP_CMDS = $(VBOX_RECOMPILER_OP_GCC_PLD_COMPAT) \
 	-S -s \
-	$(filter-out -g -O0, \
+	$(filter-out -g -O0 -fno-stack-protector, \
 		$($(REM_MOD)_CFLAGS) $($(REM_MOD)_CFLAGS.$(KBUILD_TYPE)) $($(REM_MOD)_CFLAGS.$(KBUILD_TARGET)) $($(REM_MOD)_CFLAGS.$(KBUILD_TARGET_ARCH)) $($(REM_MOD)_CFLAGS.$(KBUILD_TARGET).$(KBUILD_TARGET_ARCH)) \
 		$(target-i386/op.c_CFLAGS) $(target-i386/op.c_CFLAGS.$(KBUILD_TARGET)) $(target-i386/op.c_CFLAGS.$(KBUILD_TARGET_ARCH)) $(target-i386/op.c_CFLAGS.$(KBUILD_TARGET).$(KBUILD_TARGET_ARCH)) \
 		) \
