--- VirtualBox-4.1.6_OSE/src/VBox/Installer/linux/VBoxSysInfo.sh.orig	2011-11-04 12:19:43.000000000 +0100
+++ VirtualBox-4.1.6_OSE/src/VBox/Installer/linux/VBoxSysInfo.sh	2011-11-04 21:34:53.817325782 +0100
@@ -38,6 +38,11 @@
     then
         # LSB-compliant system
         print_linux_info `lsb_release -i -s` `lsb_release -r -s`
+    elif [ -r /etc/pld-release ] && grep PLD /etc/pld-release >/dev/null 2>&1
+    then
+	# PLD-based
+	release=`cat /etc/pld-release | sed -e 's/.PLD.*//'`
+	print_linux_info "PLD" $release
     elif [ -r /etc/debian_version ]
     then
         # Debian-based system
