diff -urN VirtualBox-2.0.0.orig/Config.kmk VirtualBox-2.0.0/Config.kmk
--- VirtualBox-2.0.0.orig/Config.kmk	2008-09-03 20:16:30.000000000 +0200
+++ VirtualBox-2.0.0/Config.kmk	2008-09-04 20:06:51.000000000 +0200
@@ -1694,10 +1694,10 @@
 else
  VBOX_PATH_QT_HOST ?= $(QTDIR)
 endif
-VBOX_UIC      ?= $(VBOX_PATH_QT_HOST)/bin/uic$(HOSTSUFF_EXE)
-VBOX_MOC      ?= $(VBOX_PATH_QT_HOST)/bin/moc$(HOSTSUFF_EXE)
-VBOX_LUPDATE  ?= $(VBOX_PATH_QT_HOST)/bin/lupdate$(HOSTSUFF_EXE)
-VBOX_LRELEASE ?= $(VBOX_PATH_QT_HOST)/bin/lrelease$(HOSTSUFF_EXE)
+VBOX_UIC      ?= /usr/bin/uic-qt4$(HOSTSUFF_EXE)
+VBOX_MOC      ?= /usr/bin/moc-qt4$(HOSTSUFF_EXE)
+VBOX_LUPDATE  ?= /usr/bin/lupdate-qt4$(HOSTSUFF_EXE)
+VBOX_LRELEASE ?= /usr/bin/lrelease-qt4$(HOSTSUFF_EXE)
 
 ifeq ($(KBUILD_TARGET),os2)
  # search DLLs in the EXE directory first (unless VBOX_DLL_QT is set)
@@ -2686,7 +2686,7 @@
 TEMPLATE_VBOXQTGUIEXE = VBox Qt GUI Executable
 TEMPLATE_VBOXQTGUIEXE_DEFS = IN_RING3 QT_NO_DEBUG QT_THREAD_SUPPORT $(ARCH_BITS_DEFS)
 TEMPLATE_VBOXQTGUIEXE_INCS = \
-	$(VBOX_PATH_QT)/include
+	/usr/include/qt4
 
 ifeq ($(KBUILD_TARGET),win)
  # drag in library configuration (we need QMAKE_PRL_DEFINES)
