--- configure.old	2015-10-02 14:46:38.000000000 +0200
+++ configure	2015-10-19 21:12:04.440451289 +0200
@@ -1591,20 +1591,18 @@
           cnf_append "PATH_SDK_QT4_LIB" '$'"(firstword `strip_L "$LIBQT4"`)"
           if [ "$foundqt4" = "2" ]; then
             cnf_append "VBOX_WITH_QT4_SUN" "1"
-          elif [ "$foundqt4" = "3" ]; then
-            cnf_append "VBOX_PATH_QT4" "$TOOLQT4"
-            cnf_append "PATH_SDK_QT4" "$TOOLQT4"
-            cnf_append "PATH_TOOL_QT4_BIN" "$TOOLQT4BIN"
-            return
+          fi
+          if ! [ "$foundqt4" = "3" ]; then
+            TOOLQT4BIN="$TOOLQT4/bin"
           fi
           test_header "Qt4 devtools"
           # try it with a suffix, some platforms use that
-          if which_wrapper "$TOOLQT4/bin/moc-qt4" > /dev/null; then
+          if which_wrapper "$TOOLQT4BIN/moc-qt4" > /dev/null; then
             QT4BINSUFF="-qt4"
           else
             QT4BINSUFF=""
           fi
-          moc_ver=`$TOOLQT4/bin/moc$QT4BINSUFF -v 2>&1|sed 's+^.*(Qt \(.*\))+\1+'`
+          moc_ver=`$TOOLQT4BIN/moc$QT4BINSUFF -v 2>&1|sed 's+^.*(Qt \(.*\))+\1+'`
           if [ $? -ne 0 ]; then
             log_failure "moc$QT4BINSUFF not working"
             fail
@@ -1612,7 +1610,7 @@
             log_success "found version $moc_ver"
             cnf_append "VBOX_PATH_QT4" "$TOOLQT4"
             cnf_append "PATH_SDK_QT4" "$TOOLQT4"
-            cnf_append "PATH_TOOL_QT4_BIN" "$TOOLQT4/bin"
+            cnf_append "PATH_TOOL_QT4_BIN" "$TOOLQT4BIN"
             [ -n "$QT4BINSUFF" ] && cnf_append "TOOL_QT4_BIN_SUFF" "$QT4BINSUFF"
           fi
         fi
