--- configure.orig	2007-01-16 11:39:32.000000000 +0100
+++ configure	2007-01-18 11:20:42.000000000 +0100
@@ -51,7 +51,6 @@
 LIBZ="-lz"
 INCPNG=""
 LIBPNG="-lpng"
-QTDIR="/usr/qt/3 /usr/lib/qt3 /usr/share/qt3"
 KBUILDDIR="`cd $(dirname $0); pwd`/kBuild"
 DEVDIR="`cd $(dirname $0); pwd`/tools"
 if [ -d /lib/modules/`uname -r`/build ]; then
@@ -148,8 +147,8 @@
   echo "compiling the following source file:" >> $LOG
   cat .tmp_src.cc >> $LOG
   echo "using the following command line:" >> $LOG
-  echo "$CXX -O -Wall -o .tmp_out .tmp_src.cc \"$1\"" >> $LOG
-  $CXX -O -Wall -o .tmp_out .tmp_src.cc $1 >> $LOG 2>&1
+  echo "$CXX -O -Wall -o .tmp_out .tmp_src.cc -L/usr/X11R6/lib \"$1\"" >> $LOG
+  $CXX -O -Wall -o .tmp_out .tmp_src.cc -L/usr/X11R6/lib $1 >> $LOG 2>&1
   if (($?!=0)); then
     if [ -z "$4" ]; then
       echo -e "\n  $2 not found at $1 or $3 headers not found"
@@ -873,29 +873,25 @@
 }
 ' > .tmp_src.cc
   found_qt=0
-  for q in $QTDIR; do
-    echo "compiling the following source file:" >> $LOG
-    cat .tmp_src.cc >> $LOG
-    echo "using the following command line:" >> $LOG
-    echo "$CXX -O -Wall -o .tmp_out .tmp_src.cc -I$q/include -L$q/lib -lqt-mt" >> $LOG
-    $CXX -O -Wall -o .tmp_out .tmp_src.cc -I$q/include -L$q/lib -lqt-mt >> $LOG 2>&1
-    if (($?==0)); then
-      if test_execute; then
-        cnf_append "QTDIR" `cd $q ; pwd`
-        found_qt=1
-        break
-      fi
+  echo "compiling the following source file:" >> $LOG
+  cat .tmp_src.cc >> $LOG
+  echo "using the following command line:" >> $LOG
+  echo "$CXX -O -Wall -o .tmp_out .tmp_src.cc -I/usr/include/qt -L/usr/lib -lqt-mt" >> $LOG
+  $CXX -O -Wall -o .tmp_out .tmp_src.cc -I/usr/include/qt -L/usr/lib -lqt-mt >> $LOG 2>&1
+  if (($?==0)); then
+    if test_execute; then
+      found_qt=1
     fi
-  done
+  fi
   if (($found_qt!=1)); then
-    echo -e "\n  Qt not found at \"$QTDIR\" or Qt headers not found"
+    echo -e "\n  Qt not found or Qt headers not found"
     echo      "  Check the file $LOG for detailed error information."
     fail
     return 1
   fi
   test_header "Qt devtools"
-  if check_avail "$q/bin/moc" QTDIR/bin; then
-    moc_ver=`$q/bin/moc 2>&1 -v|sed 's+^.*(Qt \(.*\))+\1+'`
+  if check_avail "/usr/bin/moc"; then
+    moc_ver=`/usr/bin/moc 2>&1 -v|sed 's+^.*(Qt \(.*\))+\1+'`
     if (($?!=0)); then
       log_failure "not found"
       fail
@@ -937,7 +932,6 @@
   --with-iasl=PATH        the position of the iasl compiler [$IASL]
   --with-linux=DIR        Linux kernel source directory [$LINUX]
   --with-mkisofs=PATH     position of mkisofs [$MKISOFS]
-  --with-qt-dir=DIR       directory for QT headers/libraries [$QTDIR]
   --with-xalan=LIB        position of the xalan library [$LIBXALAN]
   --with-xerces=LIB       position of the xerces library [$LIBXERCES]
 
@@ -970,9 +964,6 @@
     --with-kbuild=*)
       KBUILDDIR=`echo $option | cut -d'=' -f2`
       ;;
-    --with-qt-dir=*)
-      QTDIR=`echo $option | cut -d'=' -f2`
-      ;;
     --with-iasl=*)
       IASL=`echo $option | cut -d'=' -f2`
       ;;
