--- VirtualBox-4.1.0_OSE/src/VBox/Additions/common/crOpenGL/fakedri_drv.c~	2011-06-30 18:33:33.000000000 +0200
+++ VirtualBox-4.1.0_OSE/src/VBox/Additions/common/crOpenGL/fakedri_drv.c	2011-07-19 21:58:45.900949411 +0200
@@ -37,10 +37,10 @@
 
 //@todo this could be different...
 #ifdef RT_ARCH_AMD64
-# define DRI_DEFAULT_DRIVER_DIR "/usr/lib64/dri:/usr/lib/dri:/usr/lib/x86_64-linux-gnu/dri"
-# define DRI_XORG_DRV_DIR "/usr/lib/xorg/modules/drivers/"
+# define DRI_DEFAULT_DRIVER_DIR "/usr/lib64/xorg/modules/dri:/usr/lib/xorg/modules/dri:/usr/lib/xorg/modules/x86_64-linux-gnu/dri"
+# define DRI_XORG_DRV_DIR "/usr/lib64/xorg/modules/drivers/"
 #else
-# define DRI_DEFAULT_DRIVER_DIR "/usr/lib/dri:/usr/lib/i386-linux-gnu/dri"
+# define DRI_DEFAULT_DRIVER_DIR "/usr/lib/xorg/modules/dri:/usr/lib/xorg/modules/i386-linux-gnu/dri"
 # define DRI_XORG_DRV_DIR "/usr/lib/xorg/modules/drivers/"
 #endif
 
