--- VirtualBox-4.3.26/configure~	2015-03-28 22:23:14.000000000 +0200
+++ VirtualBox-4.3.26/configure	2015-03-28 22:26:31.627419402 +0200
@@ -1943,7 +1943,7 @@
   fi
   for p in $PYTHONDIR; do
     for d in $SUPPYTHONLIBS; do
-      for b in lib/x86_64-linux-gnu lib/i386-linux-gnu lib64 lib/64 lib; do
+      for b in lib/x86_64-linux-gnu lib/i386-linux-gnu lib64 libx32 lib/64 lib; do
         echo "compiling the following source file:" >> $LOG
         cat $ODIR.tmp_src.cc >> $LOG
         echo "using the following command line:" >> $LOG
