--- VirtualBox-6.1.6/src/VBox/Additions/x11/VBoxClient/display-svga-x11.cpp.orig	2020-04-09 19:34:49.000000000 +0200
+++ VirtualBox-6.1.6/src/VBox/Additions/x11/VBoxClient/display-svga-x11.cpp	2020-04-17 00:05:01.199999966 +0200
@@ -293,7 +293,7 @@
     }
 }
 
-int x11MonitorThreadFunction(RTTHREAD hThreadSelf, void *pvUser)
+DECLCALLBACK(int) x11MonitorThreadFunction(RTTHREAD hThreadSelf, void *pvUser)
 {
     (void)hThreadSelf;
     (void*)pvUser;
